packages/dds/tree/src/test/simple-tree/api/schemaCreationUtilities.spec.ts

```diff
@@ -235,4 +235,99 @@ describe("schemaCreationUtilities", () => {
 			const _b: InsertableImplicit = new DayNodes.Tomorrow();
 		}
 	});
+
+	// Highest priority issue
+	it("enum interop - adaptEnum with explicit schema", () => {
+		enum Day {
+			Today = "today",
+			Tomorrow = "tomorrow",
+		}
+
+		function getDay(value: string): Day {
+			switch (value) {
+				case "today":
+					return Day.Today;
+				case "tomorrow":
+					return Day.Tomorrow;
+				default:
+					throw new Error(`Invalid value: ${value}`);
+			}
+		}
```

Comment:
If you are using typescript enums, they can provide this conversion for you:

```suggestion
		function getDay(value: string): Day {
			if (value in Day) {
				return Day[value as keyof typeof Day];
			}
			throw new Error(`Invalid value: ${value}`);
		}
```

Reply:
Wow this is nice